extensions/wikia/Recirculation/js/incontent.js

```diff
@@ -0,0 +1,187 @@
+/*global require*/
+require([
+	'jquery',
+	'wikia.window',
+	'wikia.abTest',
+	'wikia.nirvana',
+	'wikia.mustache',
+	'ext.wikia.recirculation.tracker',
+	'ext.wikia.recirculation.utils'
+], function ($, w, abTest, nirvana, Mustache, tracker, utils) {
+	// Currently only showing for English communities
+	if (w.wgContentLanguage !== 'en') { return; }
+
+
+	var $container = $('#mw-content-text'),
+		template;
+
+	function injectInContentWidget($container) {
+		var $links = $container.find('a'),
+			sections = buildSections($container.find('h2')),
+			width,
+			firstSuitableSection,
+			topTitles;
+
+		// If this page doesn't have enough content (either links or sections) we
+		// don't want to show this widget
+		if ($links.length < 8 || sections.length < 3) {
+			return;
+		}
+
+		// The idea is to show links above the first section under an infobox
+		width = $container.outerWidth();
+		firstSuitableSection = sections.find(function(item, index) {
+			return item.width === width;
+		});
+
+		if (!firstSuitableSection) {
+			return;
+		}
+
+		topTitles = findTopTitles($links);
+
+		if (topTitles.length < 3) {
+			return;
+		}
+
+		nirvana.sendRequest({
+			controller: 'ArticlesApi',
+			method: 'getDetails',
+			format: 'json',
+			type: 'get',
+			data: {
+				titles: topTitles.join(','),
+				abstract: 0,
+				width: 50,
+				height: 50
+			},
+			callback: renderArticles(firstSuitableSection)
+		});
+	}
+
+	function renderArticles(section) {
+		var placeholderImage = w.wgExtensionsPath + '/wikia/Recirculation/images/placeholder.png';
```

Comment:
Could you consider changing this approach a little bit? There is [a practice in Wikia that we create CSS classes with images as background](https://github.com/Wikia/app/search?utf8=%E2%9C%93&q=%2F*+%24wgCdnStylePath+*%2F) and the SASS files are the only ones with the  file paths. Then in JS you just add or remove the class with placeholder image. I'd just remove the `placeholderImage` from here and set `item.thumbnail` to `null` if image does not exist. Then in the mustache template only if the `thumbnail` is *not* `null` I'd put `<img src="{{thumbnail}}" />`, otherwise I'd left `div` with a `placeholder` class which has the placeholder image defined in the style rule. Think about it :-) :art: